### ISSUE TITLE
Unexpected inference in gomodel:5f46c3b700004144

### ISSUE THREAD

**Issue by malcolmfisher103** [geneontology/go-ontology#20258]
I made a small [GO-CAM model](http://noctua.geneontology.org/editor/graph/gomodel:5f46c3b700004144) in Noctua for Xenopus fam183b from a paper that covered both Xenopus and mouse, MGI were already curating the mouse stuff. ![gomodel_5f46c3b700004144](https://user-images.githubusercontent.com/23216004/97919377-b3a71880-1d25-11eb-9c0f-f3b252acb2d0.png)

One of the annotations is for 'epithelial cilium movement involved in extracellular fluid movement [GO:0003351]'. The reasoner adds an inferred process of 'establishment of localization in cell [GO:0051649]'. This doesn't seem to be an obvious inference. I thought initially it might be because I said the process occurred in multiciliated cells but even when I removed that relationship the inference was still there.

Is this inference correct?

**Comment by pgaudet** [comment]
@vanaukenk @ukemi

**Comment by ukemi** [comment]
The inference is correct based on all of the rules. But it points to a problem in the ontology. 

'epithelial cilium movement involved in extracellular fluid movement'  is asserted to be a subclass of 'extracellular transport'.
'extracellular transport' is a subclass of 'transport'.
'transport' is a subclass of 'establishment of localization'.
The 'epithelial cilium movement involved in extracellular fluid movement'  in your model occurs_in some 'multiciliated epidermal cell'. 
'multiciliated epidermal cell is a subclass of cell.
Therefore 'epithelial cilium movement involved in extracellular fluid movement' is a type of 'transport' that occurs in a cell.
Therefore it is a type of 'establishment of localization' that occurs_in a cell.
'establishment of localization in cell' is defined as 'establishment of localization' and ('occurs in' some cell).
Following this logic, if the 'epithelial cilium movement involved in extracellular fluid movement' occurs in a cell, then it is 'establishment of localization in cell'.

I would question whether 'epithelial cilium movement involved in extracellular fluid movement'  is a subclass of 'extracellular transport'. The 'involved in' phrasing of the term suggests that when it was created, the movement of the cilium was part_of 'extracellular transport'. This would be the easiest solution, but brings up some interesting issues about the spatial relations of transport. It would be fun to look at logically defining the term with nested expressions.

'epithelial cilium movement involved in extracellular fluid movement' = ('cilium movement' occurs_in some 'epithelial cell') and results_in_movement_of some (fluid? adjacent_to some 'epithelial cell'). Or a new cc term extracellular fluid?? I can't find fluid in an ontology. It would be nice to indicate the same instance of the epithelial cell.

**Comment by malcolmfisher103** [comment]
Thanks for that detailed answer David. One small thing is that the inference persisted even when I removed the 'occurs in' relationship to the cell type. So the singular term seems to be enough to draw the inference, perhaps because the cilium 'is a' cellular component?

**Comment by ukemi** [comment]
Ah yes. I suspect that once the occurs_in is removed, the inference is because 
'cilium' is a subclass of 'plasma membrane bound cell projection'.
'plasma membrane bound cell projection' is part of a 'cell'.
'cilium movement' results in movement of some 'cilium'.
@balhoff do you agree?

**Comment by deustp01** [comment]
I'm hanging up on the occurs_in part. If a cilium protruding outwards from a cell beats and thereby sets up a flow of fluid in the extracellular space (e.g., to clear mucus from lung epithelia, or to set up a directional flow sensed by other cells in a developing embryo to establish left - right asymmetry), then even though a cell part is doing the agitating and definitely depends on things inside the cell to provide energy and leverage, the flow is happening entirely on the outside. Is there any mileage in this?

**Comment by ukemi** [comment]
Yes. There is something funny about this and the way we represent it.

**Comment by deustp01** [comment]
Drag Karen into the inside - outside discussion. She really understands the weedy details!


________________________________

**Comment by ukemi** [comment]
@malcolmfisher103 , are you sure you refreshed after you deleted the occurs_in 'multiciliated epithelial cell'? I just tried it with @balhoff and we didn't see the inference.

**Comment by ukemi** [comment]
@krchristie anything to add? An interesting discussion about cilia, where they are moving and location of the things that are a result of their movement. Perhaps bring this up on an ontology call?

**Comment by malcolmfisher103** [comment]
Hi David, you are right I hadn't refreshed it, I think I expected the reasoner to update it when I saved the changes.

**Comment by ukemi** [comment]
I'm kind of surprised it didn't when you saved. Weird.

**Comment by krchristie** [comment]
> Ah yes. I suspect that once the occurs_in is removed, the inference is because
> 'cilium' is a subclass of 'plasma membrane bound cell projection'.
> 'plasma membrane bound cell projection' is part of a 'cell'.
> 'cilium movement' results in movement of some 'cilium'.
> @balhoff do you agree?

I think it is a tautology to say that **cilium movement** results in movement of itself, and probably not really what we want in GO.

Here are the logical definitions of **cilium movement** and **microtubule-based movement**

**cilium movement (GO:0003341)** 
```
'movement of cell or subcellular component'
 and ('results in movement of' some cilium)
 and ('process has causal agent' some microtubule)

```

**microtubule-based movement (GO:0007018)**
```
'movement of cell or subcellular component'
 and ('process has causal agent' some microtubule)
```

I think that the logical definition of **cilium movement** should probably be changed to say that the cilium is the causal agent. I don't think it's correct to say that the microtubule is the causal agent of cilium movement anyway, as the MTs are present in non-motile cilia too. For cilia, the causal agent, or agents, would include the dynein arms. I'm wondering if it's really correct to say that the causal agent of **microtubule-based movement** is the MTs. It seems to me that the MTs are often the highway upon which motors move.

I'll make an ontology ticket about these logical defs.

**Comment by krchristie** [comment]
@malcolmfisher103 - Regarding your model, it seems that you should be able to put the two GO terms in line with each other since we know that cilium assembly is upstream of cilium motility. Basically, whatever Fam183b.L is doing in the process of `cilium assembly` is upstream of `epithelial cilium motility involved in extracellular fluid movement`.

**Comment by malcolmfisher103** [comment]
@krchristie I had been considering making cilium assembly upstream of the motility term, thanks for the advice.

I think David and Jim have identified the 'occurs in' relationship with the 'multiciliated epidermal cell' as the culprit for where the inference came from, with some issues with my interaction with Noctua being why it persisted. So is this an incorrect relationship that I should just remove? The cilium is definitely part of the cell but should I say the motility 'occurs in' the cell?

**Comment by krchristie** [comment]
> @krchristie I had been considering making cilium assembly upstream of the motility term, thanks for the advice.
> 
> I think David and Jim have identified the 'occurs in' relationship with the 'multiciliated epidermal cell' as the culprit for where the inference came from, with some issues with my interaction with Noctua being why it persisted. So is this an incorrect relationship that I should just remove? The cilium is definitely part of the cell but should I say the motility 'occurs in' the cell?

As a biologist, I think that the cilium movement is occurring within the cell since there are motors moving within the cilium that create the movement of the organelle as a whole and that movement is occurring within a part of the cell. However, due to its location on the external surface of the cell, the cilium is also moving through the extracellular region and the movement of extracellular fluid is definitely occurring outside of the cell. Many things about the cilium are tricky in the ontology. It isn't the first time we've had a hard time defining cilium terms in the ontology. I've added the editors-discussion tag so that we'll discuss this at an ontology editors' call.

**Comment by ValWood** [comment]
The problem is the

GO:0003351    epithelial cilium movement involved in extracellular fluid movement
parent
GO:0099111    microtubule-based transport

movement is not always transport.
This parentage is wrong on a number of levels.
The fluid is being transported by the epithelia.
cillium movement is a microtuble based movement, but not a microtubule based transport
I guess this term would need an additional parent to describe the fluid movement

**Comment by ValWood** [comment]
> movement is not always transport.

was an understatement.

Movement and transport are different things, but often conflated in the ontology.
"movement is physical motion between points in space while transport is an act of transporting"

microtubules move things (i.e  as cillia), or  by pushing (the nucleus), or pulling (chromosomes apart). I don't think this is transport?

they are also part_of the process of transport, when they provide the tracks for vesicles and cargo to be moved (for instance components of the spindle pole body,  are transported along the spindle  microtubules towards the SPB). Lost chromsomes are pulled toward the spindle pole body during mitosis. I think this happens on 'depolymerizing mnicrotubules', so it might look a bit like transport, but it probably isn't.

I don't know if this helps to disambiguate?

At present we have 
GO:0099111   microtubule-based transport
as part of transport, but not all microtubule based movement is transport.

I think you  might fix the problem here by moving

GO:0003351 epithelial cilium movement involved in extracellular fluid movement

to be a child of
GO:0007018   microtubule-based movement

instead of
GO:0099111   microtubule-based transport

**Comment by pgaudet** [comment]
Maybe the genera should be 'GO:0046907 intracellular transport' 
rather than 'movement of cell or subcellular component'

(or another term - just 'movement of cell or subcellular component' is a strange grouping class.

**Comment by krchristie** [comment]
**11/9/20 - Discussion on ontology editors call**
- The logical definition of `microtubule-based movement (GO:0007018)` = `'movement of cell or subcellular component' and ('process has causal agent' some microtubule)` is not sufficient. It also requires an additional causal agent of some type of microtubule motor.
- In the logical definition of `cilium movement (GO:0003341)`, it seems very odd that the genera is a term that is so broad: `'movement of cell or subcellular component'`. @krchristie will look into what it should be.

**Comment by krchristie** [comment]
> Maybe the genera should be 'GO:0046907 intracellular transport'
> rather than 'movement of cell or subcellular component'

I don't think that `'GO:0046907 intracellular transport'` will work since the microtubule motors within the cilium don't transport anything; they just change the relative positions of the microtubules within the cilium resulting in cilium bending, aka movement.

> (or another term - just 'movement of cell or subcellular component' is a strange grouping class.

Agreed!

**Comment by ukemi** [comment]
Is that always true? I thought there were cases where there was transport of receptors along microtubules in the cilium so that they could be deposited in the cilium and the cilium would act as an antenna for the ligands.

**Comment by krchristie** [comment]
> Is that always true? I thought there were cases where there was transport of receptors along microtubules in the cilium so that they could be deposited in the cilium and the cilium would act as an antenna for the ligands.

Transport of receptors along MTs in the cilium occurs by IFT (intraflagellar transport) and is caused by movement of the IFT motors along ciliary MTs, so this is transport. This occurs in non-motile primary cilia for sure, and probably also in motile cilia, but I haven't seen much research characterizing signaling in motile cilia. IFT motors do not generate movement of cilia, though IFT is required upstream of ciiary movement because IFT is required to transport things into the cilium as part of assembly, so when IFT is broken, ciliary movement is generally also broken.

Ciliary movement is caused by the motors of the dynein arms, which are present in motile cilia, but not in non-motile ones. Movement of dynein arms does not transport anything, but shifts the relative connection points between two ciliary MTs such that bending of the cilium occurs.

I edited my previous comment to accurately specify `the microtubule motors THAT MOVE the cilium` rather than the `the microtubule motors IN the cilium`

**Comment by krchristie** [comment]
> microtubules move things (i.e as cillia), or by pushing (the nucleus), or pulling (chromosomes apart). I don't think this is transport?

MTs are often the scaffold along which a MT-motor may move along a MT to generate movement; this is how motile cilia move and a common mechanism of transport. However, MTs alone are often not sufficient for movement, e.g. non-motile cilia which have MTs, but no dynein arms with motors to shift the relative positioning of the MTs. 

My recollection is that there are other types of microtubule mediated movement where the movement is generated by shortening or lengthening an anchored MT that has some cargo attached at other end. @ValWood - do you know if this is what is occurring in the examples you mentioned above for "**pushing (the nucleus)**, or **pulling (chromosomes apart)**"?

**Comment by krchristie** [comment]
I see that we already have this term: **microtubule polymerization based movement (GO:0099098)**. But this term **chromosome movement towards spindle pole (GO:0051305)** is due to shortening rather than polymerization so it isn't a child of that term.

I think we're going to need some new grouping terms under `microtubule movement`. Here's some initial thoughts.

-- microtubule movement (GO:0007018)
--- term to group movement based on MT length change (GO:NEW)
---- microtubule depolymerization based movement (GO:NEW)
---- microtubule polymerization based movement (GO:0099098)
--- term to group movement based on movement of motor along MT (GO:NEW)

not sure whether existing term `microtubule sliding` is related to motors or not...

We may also need new child terms under `microtubule-based transport` to reflect the subgroupings I'm thinking about above.

**Comment by ValWood** [comment]
Yes, for chromsome pulling apart microtubules depolymerize, and the DAM/DASH complex acts as a rachet, so this is the microtubule kinetochore attachment.
https://www.mdpi.com/1422-0067/20/24/6182/htm
This is 'microtubule-dependent movement', but it isn't transport (at least, I do not think anybody would describe chromosome segregation as a type of transport, but the chromosomes move apart).

Then there is this situation:
https://www.pombase.org/reference/PMID:31483748
microtubules-dependent processes also  provide the force to oscillate the nucleus at interphase (I don't think this is transport intuitively either).

There is also  the microtubule-based movement  with kinesins providing the force moving the SPBs apart  during spindle formation
https://www.pombase.org/reference/PMID:29167352

These processes are all microtubule-based movement, but not transport. I looked at these terms last night and they all seemed to make sense except for the one parentage I pointed out above.

I think this proposal  is making things more complicated. I wouldn't add any more grouping terms, I also wouldn't put depolymerization in the process terms since ideally these would be in the molecular function ontology. We have been trying to move these recently because we now more about microtubule  polymerases and the depolymerases. I would like to remove "microtubule polymerization based movement" and "microtubule sliding" from BP because these include  molecular functions in the processes and would require an exponential number of terms to make these work for all processes.

There are quite a few open tickets about this. Adding more processes which include  activities isn't going to help,  because there are so many mechanisms at play.

 I still think you can resolve the current problem by removing

GO:0003351 epithelial cilium movement involved in extracellular fluid movement
parent
GO:0099111 microtubule-based transport

which is clearly wrong. Cillium movement isn't microtubule-based transport.

**Comment by ValWood** [comment]
I updated the above comments, I think the depolymerization terms are OK but I'm sure there are now aspects of these processes, especially around chromosomes segregation, where we know a lot of the activities and the processes could be simplified to allow more modular annotation.

But I don't think it is simple to classify as polymerization or depolymerization based movement. A single process can combine a number of mechanisms and forces.

**Comment by ValWood** [comment]
I requested some of these polymerization terms over the years when  we knew less about the processes.

**Comment by krchristie** [comment]
> I still think you can resolve the current problem by removing
> 
> GO:0003351 epithelial cilium movement involved in extracellular fluid movement
> parent
> GO:0099111 microtubule-based transport
> 
> which is clearly wrong. Cillium movement isn't microtubule-based transport.

It really isn't quite that simple.

This parentage is coming from reasoning based on the existing equivalence axiom of several terms including this term:

**cilium movement (GO:0003341)** 
```
'movement of cell or subcellular component'
 and ('results in movement of' some cilium)
 and ('process has causal agent' some microtubule)
```

I've already commented about the fact that this equivalence axiom is not sufficient. Fixing this may require adding grouping terms under microtubule-based movement in order to make appropriate logical axioms for other terms.

Then this term, the one being used for annotation,

**epithelial cilium movement involved in extracellular fluid movement (GO:0003351)**
does not have an equivalence axiom, but does have these **ASSERTED** SubClass Of relationships:
- cilium movement
- extracellular transport [def = The transport of substances that occurs outside cells.]

and this **INFERRED** SubClass Of relationship:
- microtubule-based transport

The ASSERTED relationships seem true to, but the INFERRED one is incorrect. To remove the inferred relationship, it is likely that we will have to fix the logic in microtubule-based movement, which I hope will make it possible for me to define this term with an equivalence axiom, rather than just with some directly asserted relationships.

**Comment by ValWood** [comment]
OK got it.

I still think we should ask first whether "microtubule-based movement" is really a useful grouping term.
All of the children should have other parentage describing specific processes, and "microtubule-based movement" groups so many unrelated concepts (other than that they are microtubule-based)  it might not really be so useful (microtubule-based process would suffice).

Also, the persistance of the precomposed term-  I thought GO was  removing all of the precomposed  process terms with  "involved in" so the speciific term epithelial cilium movement involved in extracellular fluid movement (GO:0003351) presumably would go at some point?

**Comment by pgaudet** [comment]
I dont know very much about those processes, but it seems to me 'microtubule-based movement (GO:0007018)' is a good grouping term, and can include microtubule polymerization and depolymerization-based movements, but I am wary of creating too many grouping therms under that class - so I would NOT create those. In fact I suggest we go the other way sand obsolete 'microtubule polymerization based movement (GO:0099098) (has no direct annotations). 


I suggest we dont create those: 

> --- term to group movement based on MT length change (GO:NEW)
The definition of 'microtubule-based movement (GO:0007018) already covers that: "A microtubule-based process that results in the movement of organelles, other microtubules, or other cellular components. Examples include motor-driven movement along microtubules and movement driven by polymerization or depolymerization of microtubules."

>---- microtubule depolymerization based movement (GO:NEW)
--- term to group movement based on movement of motor along MT (GO:NEW)

Thanks, Pascale

**Comment by ValWood** [comment]
I agree.  'microtubule-based movement (GO:0007018) is sufficient. The rest are mechanistic, so they now seem odd in BP.

**Comment by krchristie** [comment]
The reason I'm suggesting these terms is that I think that they may be needed in order to make equivalence axioms that are both necessary and sufficient. For some of these types of MT-based movement, a MT motor that moves along the MT is required. I don't think that is true for the types of movement that are based on MT length change, i.e. polymerization or depolymerization. I don't want them as annotation terms, but as logical definitions that can be used in order to define other terms.

**Comment by pgaudet** [comment]
Let's see if we can do without, first ?

**Comment by raymond91125** [comment]
**[Written by Claude]**

Cross-referencing #20574, where this is now being triaged. @krchristie's highway-not-agent point holds and generalises: the microtubule and actin-filament rows are 9 of the 104 causal-agent logical definitions, and they are the clearest errors in the set — distinct from the signal-carrier and machinery uses, which are mostly defensible.

One extra data point for `cilium movement` (GO:0003341) specifically — RO:0002608 sits under `causally related to`, not `has participant`, so the axiom does not even assert that the microtubule participates in the movement. Whatever replaces it should be a participation relation.

Flagging that GO:0007017 `microtubule-based process` and GO:0030029 `actin filament-based process` are branch roots, so this subset has the largest blast radius of the seven groups and should be reasoned before and after.
